### PR TITLE
rename initialize_beacon_state{_from_eth1,}; suppress warnings when doppelganger detection disabled

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: beacon_chain/gossip_processing/eth2_processor.nim [hunk 1]
```diff
@@ -172,16 +169,14 @@ proc checkForPotentialDoppelganger(
       tgtBlck, attestationData.target.epoch)
     for validatorIndex in attesterIndices:
       let validatorPubkey = epochRef.validator_keys[validatorIndex]
-      if self.validatorPool[].getValidator(validatorPubkey) !=
-          default(AttachedValidator):
-        warn "Duplicate validator detected; would be slashed",
```
Comment (arnetheduck) [inline beacon_chain/gossip_processing/eth2_processor.nim:177]: shouldn't this still print the warning iff the attestation cannot be found in the slashing protection db?

Comment (tersec) [reply beacon_chain/gossip_processing/eth2_processor.nim:177]: The slashing database integration doesn't exist yet, pending updates from @mratsim. I believe https://github.com/status-im/nimbus-eth2/pull/2452 is the relevant PR for that.

This just adjusts the behavior of the existing version, which doesn't use the slashing protection database.

**Comment by arnetheduck** [comment]
`initialize_beacon_state_from_eth1` comes from the spec though?

**Comment by tersec** [comment]
Oops, yeah: https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/beacon-chain.md#genesis

I knew at some point this had some useless appendage to its name that was meant to be temporary, but it wasn't `from_eth1`. The whole thing shouldn't be reverted, but I'll make a PR that reverts that part (while leaving the more visible doppelganger detection part).

**Pull request merged**